base/libgit2/diff.jl

```diff
@@ -36,9 +36,12 @@ function Base.count(diff::GitDiff)
 end
 
 function Base.getindex(diff::GitDiff, i::Csize_t)
+    if i < 1 || i > count(diff)
+        throw(BoundsError(diff, i))
+    end
     delta_ptr = ccall((:git_diff_get_delta, :libgit2), Ptr{Void},
                        (Ptr{Void}, Csize_t), diff.ptr, i-1)
-    delta_ptr == C_NULL && return nothing
+    delta_ptr == C_NULL && throw(Error.GitError(Error.ERROR))
```

Comment:
I think, `BoundsError` is more appropriate here. C_NULL is returned if out of bounds ([ref](https://libgit2.github.com/libgit2/#HEAD/group/index/git_index_get_byindex)). Use the same exception in similar cases for `GitIndex`, etc.